model/pdata/common.go

```diff
@@ -744,6 +744,31 @@ func (am AttributeMap) CopyTo(dest AttributeMap) {
 	*dest.orig = origs
 }
 
+// AsMap converts an OTLP AttributeMap to a standard go map
+func (am AttributeMap) AsMap() map[string]interface{} {
```

Comment:
I was thinking that we should better return the `map[string]AttributeValue` to be consistent with New..FromMap?

Reply:
Yeah I'm working through this, the only pain point is the json marshalling.

Reply:
That is true, and I have this in my mind. We have the "AsString" from AttributeValue not really producing JSON encoding, maybe we should?

Reply:
It is interesting because:

```golang
// This produces "foo"
pdata.NewAttributeValueString("foo").AsString()
// This produces "{\"foo\": \"bar\"}"
attr := pdata.NewAttributeValueMap()
attrMap := attr.Map()
attrMap.UpsertString("foo", "bar")
attr.AsString()
```

So we are inconsistent on how we encode strings in these cases :(

Reply:
Yup, the inconsistency is described here: https://github.com/open-telemetry/opentelemetry-collector/issues/3815#issuecomment-896635804

Reply:
I've created a PR to see what changing the return looks like, PTAL: https://github.com/codeboten/opentelemetry-collector/pull/1

Reply:
Looking at the use of `AttributeMapToMap` in the contrib repo, it seems the best thing to do here may be to just create a `toJSON` method as it appears to be the most common usecase

Reply:
OK. Merged the change to update the return type, added a separate private method to ensure JSON marshalling will continue to work for AttributeMap. PTAL @bogdandrutu